tests/test_sockets.py

```diff
@@ -332,10 +360,9 @@ async def test_connection_refused(
         exception_class: type[ExceptionGroup] | type[ConnectionRefusedError],
         fake_localhost_dns: None,
     ) -> None:
-        dummy_socket = socket.socket(AddressFamily.AF_INET6)
-        dummy_socket.bind(("::", 0))
-        free_port = dummy_socket.getsockname()[1]
-        dummy_socket.close()
+        with socket.socket(AddressFamily.AF_INET6) as dummy_socket:
```

Comment:
oh this should either be left alone or upgraded to free_tcp_port

Reply:
Upgrade to `free_tcp_port` if you can.

Reply:
Will do